File: Card_Game_Engine/Models/Filter/Classes/FilterCondition.cs
using Card_Game_Engine.Models.Filter.Enums;
using Card_Game_Engine.Models.Filter.Interfaces;

namespace Card_Game_Engine.Models.Filter.Classes;

public class FilterCondition : IFilterNode
{
    public PropertyEnum PropertyId { get; set; }
    public OperatorEnum Operator { get; set; }
    public string Value { get; set; }
    public LogicalOperatorEnum? LogicalOperator { get; set; }
}

File: Card_Game_Engine/ClientApp/src/app/_reusable-components/parameter/filter-modal/_namespace/classes/filter-condition.ts
import {ReferenceEnum} from "../enums/reference.enum";
import {OperatorEnum} from "../enums/operator.enum";
import {IFilterNode} from "../interfaces/filter-node";
import {LogicalOperatorEnum} from "../enums/logical-operator.enum";

export class FilterCondition implements IFilterNode {
  id: ReferenceEnum;
  operator: OperatorEnum;
  value: string;
  logicalOperator?: LogicalOperatorEnum;

  constructor(id: ReferenceEnum, operator: OperatorEnum, value: string) {
    this.id = id;
    this.operator = operator;
    this.value = value;
    this.logicalOperator = null;
  }
}
